app/models/user.rb

```diff
@@ -28,17 +30,36 @@ def new_token
     end
   end
 
+  def activate
+    update_columns activated: true, activated_at: Time.zone.now
+  end
+
+  def send_activation_email
+    UserMailer.account_activation(self).deliver_now
+  end
+
+  def authenticated?(attribute, token)
+    digest = send "#{attribute}_digest"
```

Comment:
`send` ở đây hoạt động như thế nào

Reply:
which lets us call a method with a name of our choice by “sending a message” to a given object.
method send cho phép ta gọi 1 method với tên cho trước ở cặp dấu ngoặc kép, ở đây là activation_digest. Tương đương với việc gọi ra thuộc tính của đối tượng user cho trước.